src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -30,81 +38,108 @@ public PostgresSQLConnection(DbConnection conn) {
 
   @Override
   public SQLConnection setAutoCommit(boolean autoCommit, Handler<AsyncResult<Void>> handler) {
-    throw new UnsupportedOperationException();
+    inAutoCommit.set(autoCommit);
+    if (inTransaction.compareAndSet(true, false) && autoCommit) {
```

Comment:
the boolean autocommit should be evaluated before changing the state of the `inTransaction` atomic